### PR TITLE
Login frontend 1

### PR DESCRIPTION
**Pull request by AleMathus**
### **Summary**
Implementing a view for the login screen.

### **Trello**
https://trello.com/c/mmScndmC

### **Screenshot**
image

### CODE REVIEW

File: app/src/main/res/layout/fragment_login.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            android:id="@+id/log_in_button"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/small_margin"
+            android:layout_marginTop="@dimen/small_margin"
+            android:layout_marginEnd="@dimen/small_margin"
+            android:backgroundTint="@color/white"
+            android:text="@string/log_in"
+            android:textColor="@color/black"
+            android:textSize="16sp"
```
Comment (LisandroDiMeo) [inline app/src/main/res/layout/fragment_login.xml]: This value could be inside dimens.xml

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -6,13 +6,15 @@
...
+        android:name="androidx.navigation.fragment.NavHostFragment"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:defaultNavHost="true"
         app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
-
+        app:layout_constraintLeft_toLeftOf="parent"
```
Comment (LisandroDiMeo) [inline app/src/main/res/layout/activity_main.xml]: Here you're using Left2Right constraints. But below I saw that you were using Start2End. Usually is better not to mix them and pick one alternative. there is no much difference between its functionalities though ✋

**Review by LisandroDiMeo — APPROVED** [review]
It looks great to me! 🤟 Only two small comments, and (for me) we are ready to merge

File: app/src/main/java/com/example/myapplication/ui/LoginFragment.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.example.myapplication.databinding.FragmentLoginBinding
+
+class LoginFragment: Fragment() {
+    lateinit var binding: FragmentLoginBinding
```
Comment (ffasitella) [inline app/src/main/java/com/example/myapplication/ui/LoginFragment.kt]: This property should be private.

File: app/src/main/java/com/example/myapplication/ui/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentLoginBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+}
```
Comment (ffasitella) [inline app/src/main/java/com/example/myapplication/ui/LoginFragment.kt]: Add an empty line at the end of every file :)

File: app/src/main/res/layout/fragment_login.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".MainActivity">
+
+        <ImageView
+            android:id="@+id/logo_cover"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:contentDescription="@string/description_login_cover"
+            android:src="@mipmap/login_cover"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: You shouldn't store these kind of assets in the `mipmaps` folder. Try moving them to the `drawables` folder.

File: app/src/main/res/layout/fragment_login.xml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            app:layout_constraintTop_toTopOf="parent"
+            android:scaleType="fitXY"
+            android:foreground="@drawable/gradient_background"/>
+
+        <ImageView
+            android:id="@+id/wolox_logo"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:contentDescription="@string/description_wolox_logo"
+            android:src="@mipmap/wolox_logo_large"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: This one too.

File: app/src/main/res/layout/fragment_login.xml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            android:id="@+id/sign_up_button"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/small_margin"
+            android:layout_marginTop="@dimen/small_margin"
+            android:layout_marginEnd="@dimen/small_margin"
+            android:backgroundTint="@color/wolox_blue"
+            android:text="@string/sign_up"
+            android:textColor="@color/white"
+            android:textSize="16sp"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: This one should be in dimens.xml too :)

**Review by ffasitella — CHANGES_REQUESTED** [review]
Good job! Just left some feedback to work on.
Also, remember that every PR's description should also be in english.

File: app/src/main/res/layout/fragment_login.xml (hunk 5 above)
Comment (jcruz-wolox) [reply app/src/main/res/layout/fragment_login.xml]: The mipmap folder is used to save the icon app.

**Comment by jcruz-wolox** [comment]
Very good!

I don't have much to add, fix Lisan and Fede mention to close the PR.

**Review by ffasitella — APPROVED** [review]
Excellent job! 🚀

**Review by jcruz-wolox — APPROVED** [review]

**Pull request merged**